electron.vite.config.js

```diff
@@ -2,11 +2,36 @@ import { defineConfig } from "electron-vite";
 
 import ViteYaml from "@modyfi/vite-plugin-yaml";
 
-// electron.vite.config.js
+import { resolve } from "path";
+
 export default defineConfig({
-    main: {},
-    preload: {},
+    main: {
+        build: {
+            rollupOptions: {
+                input: {
+                    index: resolve(__dirname, "src/electron/main/main.ts"),
+                },
+            },
+        },
+    },
+    preload: {
+        build: {
+            rollupOptions: {
+                input: {
+                    index: resolve(__dirname, "src/electron/preload/preload.js"),
+                },
+            },
+        },
+    },
     renderer: {
+        root: "./src/electron/renderer",
         plugins: [ViteYaml()],
+        build: {
+            rollupOptions: {
+                input: {
+                    index: resolve(__dirname, "index.html"),
+                },
+            },
+        },
```

Comment:
```suggestion
        root: "./src/electron/renderer",
        plugins: [ViteYaml()],
        build: {
            rollupOptions: {
                input: {
                    index: resolve(__dirname, "src/electron/renderer/index.html"),
                },
            },
        },
```

@garrettmflynn I'm encountering trouble here doing a local build (`npm run build:win`)

 Without this change everything worked fine in dev mode (`npm start`)

But when I tried to build and test that build, that there was a rollup issue finding the `index.html` file - making this change fixed it and doesn't seem to break dev mode either (which only seems sensitive to the `root`)

However, once I overcome that index issue I then run into 

```bash
cannot find specified resource "electron/renderer/assets/app-icon/logo-guide-draft.ico", nor relative to "D:\GitHub\nwb-guide\build", neither relative to project dir ("D:\GitHub\nwb-guide")
```

and the resulting `build` folder does not contain the `renderer` subfolder either

Any ideas?

Reply:
Looks like it was being output in the `root` location (i.e. `/src/electron/renderer/build`). Updated the configuration to explicitly output to the repo build folder.

Reply:
Full build renders well!